Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -126,6 +126,12 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                     return buffer.validate().bind { validationResult in
                         if let invalidError = validationResult.failureValue as? BufferInvalidError {
                             self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
+
+                            log.warning("Buffer inconsistent, starting repair procedure")
+                            let repairer = BookmarksRepairRequestor(scratchpad: self.scratchpad, basePrefs: self.basePrefs, remoteClients: remoteClientsAndTabs)
```

Comment:
Call `greenLight` to check whether we should begin this work. It's likely that we shouldn't start repair if we're in the background.